### ISSUE TITLE
Padding and Masking with 2D Data (to apply LSTM)

### ISSUE THREAD

**Issue by hailsong** [keras-team/keras#14593]
Hi, I'm developing lstm model with 2D input.

I checked this keras tutorials, about masking and padding (https://www.tensorflow.org/guide/keras/masking_and_padding) but I wonder how can I set input with 2 dimensional data.

At the link, input data is English word and the tutorial changed it integer data, so they can reduce dimension to 1D.
![image](https://user-images.githubusercontent.com/73508733/114643006-d5162280-9d0f-11eb-8a53-9b2022a9d807.png)

In my case, I wanna make model with input : (50, 63 array) output : 1 (classification)
because I have 336 data, after padding, I got (336, 50, 63) size of mask tensor.

**padded = pad_sequences(df, dtype = 'float64', padding = 'post')
print(padded)

embedding = layers.Embedding(input_dim=5000, output_dim=16, mask_zero=True)
masked_output = embedding(padded)

print(masked_output._keras_mask)**

![image](https://user-images.githubusercontent.com/73508733/114643172-245c5300-9d10-11eb-996b-12f1ce924b0a.png)

at the tutorial, they get mask of size (3,6)
![image](https://user-images.githubusercontent.com/73508733/114643419-93d24280-9d10-11eb-9fb0-6f807d2cea9d.png)

Here's my question.. I padded successfully with code keras.preprocessing.sequence.pad_sequences, but embedding layer can't read what data is padded with zero. 
![image](https://user-images.githubusercontent.com/73508733/114643603-f4617f80-9d10-11eb-92b6-ef6177ac200f.png)

error code:
![image](https://user-images.githubusercontent.com/73508733/114643620-fcb9ba80-9d10-11eb-834f-bf12f6ef5bcf.png)

How can the embedding layers catch multi-dimensional input and what is padded and what isn't?
Is there any way(or example) to pad and mask with 2d data?

I'm using
Keras==2.4.3
Keras-Preprocessing==1.1.2
tensorflow==2.4.1
tensorflow-estimator==2.4.0

**Comment by saichatla** [comment]
Could you post your code please?

**Comment by chunduriv** [comment]
@hailsong, Sorry for the late response. Is this still an issue for you?                                                                                     
  
Does [this reference](https://stackoverflow.com/questions/44583254/valueerror-input-0-is-incompatible-with-layer-lstm-13-expected-ndim-3-found-n) helps you? Else, can you please share standalone code to replicate your issue? Thanks!
